realm/realm-library/src/androidTest/kotlin/io/realm/DictionaryTests.kt

```diff
@@ -513,47 +674,235 @@ class DictionaryTests {
+                    put(KEY_BYE, VALUE_BYE_NUMERIC.toByte())
+                    put(KEY_NULL, null)
+                }
+            }
+
+            val dictionaryObjectFromRealm = transactionRealm.copyToRealm(dictionaryObject)
+            val dictionaryFromRealm = dictionaryObjectFromRealm.myByteDictionary
+            assertNotNull(dictionaryFromRealm)
+
+            assertEquals(VALUE_HELLO_NUMERIC.toByte(), dictionaryFromRealm[KEY_HELLO])
+            assertEquals(VALUE_BYE_NUMERIC.toByte(), dictionaryFromRealm[KEY_BYE])
+            assertNull(dictionaryFromRealm[KEY_NULL])
+        }
+    }
+
+    @Test
+    fun copyToRealm_byteArray() {
+        val byteArray1 = ByteArray(2).apply {
+            // 0 is MIN, 1 is MAX
+            set(0, Byte.MIN_VALUE)
+            set(1, Byte.MAX_VALUE)
+        }
+        val byteArray2 = ByteArray(2).apply {
+            // Opposite of hello
+            set(0, Byte.MAX_VALUE)
+            set(1, Byte.MIN_VALUE)
+        }
+        realm.executeTransaction { transactionRealm ->
+            val dictionaryObject = DictionaryClass().apply {
+                myByteArrayDictionary = RealmDictionary<ByteArray>().apply {
+                    put(KEY_HELLO, byteArray1)
+                    put(KEY_BYE, byteArray2)
+                    put(KEY_NULL, null)
+                }
+            }
+
+            val dictionaryObjectFromRealm = transactionRealm.copyToRealm(dictionaryObject)
+            val dictionaryFromRealm = dictionaryObjectFromRealm.myByteArrayDictionary
+            assertNotNull(dictionaryFromRealm)
+
+            val byteArrayFromDictionary1 = dictionaryFromRealm[KEY_HELLO]
+            assertNotNull(byteArrayFromDictionary1)
+            Arrays.equals(byteArray1, byteArrayFromDictionary1)
+            val byteArrayFromDictionary2 = dictionaryFromRealm[KEY_BYE]
+            assertNotNull(byteArrayFromDictionary2)
+            Arrays.equals(byteArray1, byteArrayFromDictionary1)
+            assertNull(dictionaryFromRealm[KEY_NULL])
+        }
+    }
+
+    @Test
+    fun copyToRealm_date() {
+        val date1 = Date()
+        val date2 = Date()
+
+        realm.executeTransaction { transactionRealm ->
+            val dictionaryObject = DictionaryClass().apply {
+                myDateDictionary = RealmDictionary<Date>().apply {
+                    put(KEY_HELLO, date1)
+                    put(KEY_BYE, date2)
+                    put(KEY_NULL, null)
+                }
+            }
+
+            val dictionaryObjectFromRealm = transactionRealm.copyToRealm(dictionaryObject)
+            val dictionaryFromRealm = dictionaryObjectFromRealm.myDateDictionary
+            assertNotNull(dictionaryFromRealm)
+
+            assertEquals(date1, dictionaryFromRealm[KEY_HELLO])
+            assertEquals(date2, dictionaryFromRealm[KEY_BYE])
+            assertNull(dictionaryFromRealm[KEY_NULL])
+        }
+    }
+
+    @Test
+    fun copyToRealm_decimal128() {
```

Comment:
Missing UUID test? Same comment as before. Maybe convert all of these to private functions being called from a "super"-method that runs through an enum-switch?

Reply:
Sure, I will refactor it in the upcoming PR.

Reply:
And no, the UUID test was already there from the previous PR - that's why it doesn't appear in the diff.